File: src/main/java/toker/warbandscripts/panel/entity/Item.java
package toker.warbandscripts.panel.entity;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "item")
public class Item {
    private Integer id;
    private String codeName;
    private String name;
    private ItemType type;

    @Id
    @Column(name = "id", nullable = false)
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    @Column(name = "code_name", nullable = false, length = 32)
    public String getCodeName() {
        return codeName;
    }

    public void setCodeName(String codeName) {
        this.codeName = codeName;
    }

    @Column(name = "name", nullable = false, length = 32)
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @ManyToOne
    @JoinColumn(name = "type", referencedColumnName = "id", nullable = false)
    public ItemType getType() {
        return type;
    }

    public void setType(ItemType type) {
        this.type = type;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Item item = (Item) o;
        return Objects.equals(id, item.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, codeName, name);
    }
}

File: src/main/java/toker/panel/entity/Item.kt
package toker.panel.entity

import javax.persistence.*

@Entity
@Table(name = "item")
class Item(
        @Id
        @Column(name = "id", nullable = false)
        var id: Int? = null,

        @Column(name = "code_name", nullable = false, length = 32)
        var codeName: String? = null,

        @Column(name = "name", nullable = false, length = 32)
        var name: String? = null,

        @ManyToOne
        @JoinColumn(name = "type", referencedColumnName = "id", nullable = false)
        var type: ItemType? = null
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as Item
        if (id != other.id) return false
        return true
    }

    override fun hashCode(): Int {
        return id ?: 0
    }
}